includes/model/class-openedx-woocommerce-plugin-enrollment.php

```diff
@@ -0,0 +1,785 @@
+        add_filter( 'manage_openedx_enrollment_posts_columns', array( $this, 'add_columns_to_list_view' ) );
+    }
+
+    /**
+     * Replace admin meta boxes
+     *
+     * @return void
+     */
+    public function replace_admin_meta_boxes() {
+        remove_meta_box( 'submitdiv', $this->post_type, 'side' );
+
+        add_meta_box( 'openedx_enrollment_request_actions', sprintf( __( '%s actions', '' ), 'Open edX Enrollment Requests' ), array( $this, 'render_actions_box' ), $this->post_type, 'side', 'high' );
+    }
+
+    /**
+     * Renders the actions box for the edit post box
+     *
+     * @return void
+     */
+    public function render_actions_box() {
+        ?>
+        <ul class="enrollment_actions submitbox">
+
+            <label for="actions-select"><?php esc_html_e( 'Choose an action...', 'wp-edunext-marketing-site' ); ?></label>
+            <li class="wide" id="actions">
+                <select name="oer_action" id="actions-select">
+                    <option value="save_no_process"><?php esc_html_e( 'Save without processing', 'wp-edunext-marketing-site' ); ?></option>
+                    <option value="oer_sync"><?php esc_html_e( 'Synchronize (pull information)', 'wp-edunext-marketing-site' ); ?></option>
+                    <option value="oer_process" selected><?php esc_html_e( 'Process request', 'wp-edunext-marketing-site' ); ?></option>
+                    <option value="oer_no_pre"><?php esc_html_e( 'Process no pre-enrollment', 'wp-edunext-marketing-site' ); ?></option>
+                    <option value="oer_force"><?php esc_html_e( 'Process --force', 'wp-edunext-marketing-site' ); ?></option>
+                    <option value="oer_no_pre_force"><?php esc_html_e( 'Process no pre-enrollment --force', 'wp-edunext-marketing-site' ); ?></option>
+                </select>
+            </li>
+
+            <li class="wide">
+                <button class="button save_order button-primary"><span><?php esc_html_e( 'Apply action', 'wp-edunext-marketing-site' ); ?></span></button>
+            </li>
+
+        </ul>
+        <?php
+    }
+
+    /**
+     * Print openedx enrollment edit metabox
+     *
+     * @param WP_Post $post Current post object.
+     */
+    public function render_enrollment_info_form( $post ) {
+
+        if ( $this->post_type !== $post->post_type ) {
+            return;
+        }
+        $post_id = $post->ID;
+
+        $course_id = get_post_meta( $post_id, 'course_id', true );
+        $email     = get_post_meta( $post_id, 'email', true );
+        $username  = get_post_meta( $post_id, 'username', true );
+        $mode      = get_post_meta( $post_id, 'mode', true );
+        $is_active = get_post_meta( $post_id, 'is_active', true );
+        $order_id  = get_post_meta( $post_id, 'order_id', true );
+
+        $new_oer = false;
+        if ( ! $course_id && ! $email && ! $username ) {
+            $new_oer = true;
+        }
+        ?>
```

Comment:
I suppose there is a better way to manage templates that get rendered to the admin pages now. The only reason for this html code being here is that I did not know better.

Reply:
I will create another class to decouple the model from the view. In the next sprint for example, we can also start using partials to render those html snippets, what do you think?